system-helper/flatpak-system-helper.c

```diff
@@ -172,6 +371,27 @@ handle_deploy (FlatpakSystemHelper   *object,
       return TRUE;
     }
 
+  user_folder = g_file_get_path (g_file_get_parent (path));
+  uid = get_connection_uid (invocation, &error);
+  for (guint i = 0; i < ongoing_pulls->len; i++)
+    {
+      FlatpakSystemHelperOngoingPull *pull_tmp;
+
+      pull_tmp = g_ptr_array_index (ongoing_pulls, i);
+      if (g_strcmp0(pull_tmp->user_folder_name, user_folder) == 0)
+        {
+          ongoing_pull = pull_tmp;
+
+	  if (umount (user_folder) == -1)
+            g_warning ("flatpak-cache: Could not unmount directory %s\n", user_folder);
+
+          if (ongoing_pull->bindfs)
+            g_subprocess_force_exit (pull_tmp->bindfs);
```

Comment:
You may want to `g_subprocess_wait()` here because `force_exit()` doesn't wait for the process to die before returinng.